docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -0,0 +1,215 @@
+able to discover peers in the network that are able or willing to provide
+application snapshots.
+The implementation of this feature would require a more complex content-based
+peer discovery mechanism.
+
+However, while relevant, strengthening p2p-based State sync should be seen as
+_orthogonal_ to the solution proposed in this ADR, which should have relevant use
+cases even for an enhanced version of State sync.
+
+
+### Manual application state transfer
+
+The proposed [ADR 083][adr083]/[ADR 103][adr103] inspired the discussions that
+led to this ADR. At first glance, their proposal might look identical to the
+solution discussed here. The main distinction is that their proposal does not
+involve installing application snapshots, but relies on node operators to
+manually synchronize the application state.
+
+More precisely, their proposal is to
+"_allow applications to start with a bootstrapped state_
+(i.e., the application has already a state before the node is started)
+_alongside an empty CometBFT instance_
+(i.e., the node's block store is empty and the state store is at genesis state)
+_using a subsystem of the state sync protocol_".
+
+Starting from an "empty CometBFT instance" is a requirement for running State
+sync, which is maintained in the current proposal.
+The mentioned "subsystem of the state sync protocol" is the light client,
+responsible for verifying the application state and for bootstrapping
+CometBFT's block store and state store accordingly.
+
+The distinction, therefore, lies in the way which the state of the application
+from a running node is transferred to the new node to be bootstrapped.
+Instead of relying on application snapshots, produced by the application but
+handled by State sync, [ADR 083][adr083]/[ADR 103][adr103] assumes that node
+operators are able to manually synchronize the application state from a running
+node (it might be necessary to stop it) to a not-yet-started fresh node.
+
+The main limitation of this approach is that it relies on the ability of node
+operators to properly synchronize the application state between two nodes.
+While experienced node operators are likely able to perform this operation in a
+proper way, we have to consider a broader set of users and emphasize that it is
+an operation susceptible to errors.
+Furthermore, it is an operation that is, by definition, application-specific:
+applications are free to manage their state as they see fit, and this includes
+how and where the state is persisted.
+Node operators would therefore need to know the specifics of each application
+in order to adopt this solution.
+
+## Decision
+
+> This section records the decision that was made.
+> It is best to record as much info as possible from the discussion that happened.
+> This aids in not having to go back to the Pull Request to get the needed information.
+
+State sync should support the bootstrap of new nodes from application snapshots
+obtained out-of-band by node operators.
```

Comment:
> State sync should support the bootstrap of new nodes from application snapshots

I would add "local" to this sentence? Otherwise, it is what State sync already does...

Reply:
Is out of band the same as local though? Local to me would mean the node had a snapshot it itself stored before. Out of band can mean it was just copied or obtanined in another manner.

Reply:
I would say snapshots are local, for the sake of this definition. In general, I would say, from point 2. below, that they are local because obtained out-of-band?

Reply:
So local in this terminology is that the snapshots are applied from something eventually present on the node. Even though it originated from somewhere else. Sorry, this is just nit picking, nothing major :D

Reply:
No, that's important. Recall that we should still pick a name for the PR :)